contrib/pg_tde/src/include/access/pg_tde_tdemap.h

```diff
@@ -78,6 +81,23 @@ extern void pg_tde_move_rel_key(const RelFileLocator *newrlocator, const RelFile
 #define PG_TDE_MAP_FILENAME			"pg_tde_%d_map"
 #define PG_TDE_KEYDATA_FILENAME		"pg_tde_%d_dat"
 
+typedef struct WALKeyCacheRec
+{
+	XLogRecPtr start_lsn;
+	XLogRecPtr end_lsn;
+
+	RelKeyData *key;
+
+	struct WALKeyCacheRec *next;
+} WALKeyCacheRec;
```

Comment:
What is the plan to handle timelines? Do they cause any issues?

Reply:
This is the thing a wanted to discuss today

Reply:
It semes like it will not.